aws-ssm-maintenancewindow/src/test/java/software/amazon/ssm/maintenancewindow/CreateHandlerTest.java

```diff
@@ -1,54 +1,137 @@
+    private static final Integer DURATION = 2;
+    private static final String NAME = "TestMaintenanceWindow";
+    private static final String WINDOW_ID = "mw-1234567890";
+    private static final ResourceModel model = ResourceModel.builder()
+            .name(NAME)
+            .allowUnassociatedTargets(ALLOWED_UNASSOCIATED_TARGETS)
+            .schedule(SCHEDULE)
+            .duration(DURATION)
+            .cutoff(CUTOFF)
+            .build();
+    private static final ResourceHandlerRequest<ResourceModel> request = ResourceHandlerRequest.<ResourceModel>builder()
+            .desiredResourceState(model)
+            .build();
+    private static final CreateMaintenanceWindowRequest createMaintenanceWindowRequest =
+            CreateMaintenanceWindowRequest.builder()
+                    .name(model.getName())
+                    .allowUnassociatedTargets(model.getAllowUnassociatedTargets())
+                    .schedule(model.getSchedule())
+                    .duration(model.getDuration())
+                    .cutoff(model.getCutoff())
+                    .build();
+
+    private CreateHandler handler;
+
     @Mock
     private AmazonWebServicesClientProxy proxy;
 
     @Mock
     private Logger logger;
 
+    @Mock
+    private ExceptionTranslator exceptionTranslator;
+
+    @Mock
+    private CreateMaintenanceWindowTranslator createMaintenanceWindowTranslator;
+
     @BeforeEach
     public void setup() {
         proxy = mock(AmazonWebServicesClientProxy.class);
         logger = mock(Logger.class);
+        exceptionTranslator = mock(ExceptionTranslator.class);
+        createMaintenanceWindowTranslator = mock(CreateMaintenanceWindowTranslator.class);
+        handler = new CreateHandler(createMaintenanceWindowTranslator, exceptionTranslator);
     }
 
     @Test
-    public void handleRequest_SimpleSuccess() {
-        final CreateHandler handler = new CreateHandler();
+    public void handleCreateRequestForSuccess() {
 
-        final ResourceModel model = ResourceModel.builder().build();
+        when(createMaintenanceWindowTranslator.resourceModelToRequest(model))
+                .thenReturn(createMaintenanceWindowRequest);
 
-        final ResourceHandlerRequest<ResourceModel> request = ResourceHandlerRequest.<ResourceModel>builder()
-            .desiredResourceState(model)
-            .build();
+        final CreateMaintenanceWindowResponse result =
+                CreateMaintenanceWindowResponse.builder()
+                        .windowId(WINDOW_ID)
+                        .build();
+
+        when(
+                proxy.injectCredentialsAndInvokeV2(
+                        eq(createMaintenanceWindowRequest),
+                        ArgumentMatchers.<Function<CreateMaintenanceWindowRequest, CreateMaintenanceWindowResponse>>any()))
+                .thenReturn(result);
+
+        final ResourceModel expectedModel = request.getDesiredResourceState();
+        expectedModel.setWindowId(WINDOW_ID);
```

Comment:
I'm not sure if this step is necessary. I think when we call handler.handleRequest, it should set the request.getDesiredResourceState.windowId to WINDOW_ID.

Reply:
Yeah Good catch thanks!!!